### PR TITLE
Fix invoice number 

### PR DESCRIPTION
**Pull request by pankaj-ag**
Targets #1377

### Description
- Added a new seeder to set the invoice number based on the filename. 
- Added a new migration for invoice number in invoice table 
- Showed invoice number on the listing invoice page. 
```
php artisan module:seed Invoice  --class=ResetInvoiceNumberTableSeeder
```

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Invoice/Database/Seeders/ResetInvoiceNumberTableSeeder.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Modules\Invoice\Database\Seeders;
+
+use Illuminate\Database\Seeder;
+use Modules\Invoice\Entities\Invoice;
+use Illuminate\Database\Eloquent\Model;
+
+class ResetInvoiceNumberTableSeeder extends Seeder
```
Comment (rathorevaibhav) [inline Modules/Invoice/Database/Seeders/ResetInvoiceNumberTableSeeder.php:9]: this looks like a one time seeder, right? In that case, let's create an issue tracking to delete this in future please. I had to clean up a lot of one-time seeder files today.

File: Modules/Invoice/Resources/views/index.blade.php [hunk 2]
```diff
@@ -1,93 +1,107 @@
...
 @section('content')
+    <div class="container" id="vueContainer">
+        <br>
+        <br>
 
-<div class="container" id="vueContainer">
-    <br>
-    <br>
+        <div class="d-flex justify-content-between mb-2">
+            <h4 class="mb-1 pb-1"> Invoices</h4>
```
Comment (rathorevaibhav) [inline Modules/Invoice/Resources/views/index.blade.php:8]:
```suggestion
            <h4 class="mb-1 pb-1">Invoices</h4>
```

File: Modules/Invoice/Resources/views/index.blade.php [hunk 3]
```diff
@@ -1,93 +1,107 @@
...
+            <span>
+                <a href="{{ route('invoice.create') }}" class="btn btn-info text-white"> Add new invoice</a>
```
Comment (rathorevaibhav) [inline Modules/Invoice/Resources/views/index.blade.php:10]:
```suggestion
                <a href="{{ route('invoice.create') }}" class="btn btn-info text-white">Add Invoice</a>
```

File: Modules/Invoice/Services/InvoiceService.php [hunk 4]
```diff
@@ -71,6 +71,8 @@ public function store($data)
         $data['receivable_date'] = $data['due_on'];
         $invoice = Invoice::create($data);
         $this->saveInvoiceFile($invoice, $data['invoice_file']);
+        // Todo: We need to update the logic for this.
```
Comment (rathorevaibhav) [inline Modules/Invoice/Services/InvoiceService.php:74]: let's add a bit more detailing here as in what needs to be done in the to-do? Or maybe what is wrong with the current approach? I'm not sure when we will come to that, so just this line may not make a lot of sense to someone reading in future.

For example:
```php
// Todo: We need to update the logic to set invoice numbers. It should get 
// generated using a combination of invoice id, project id, and client id.
```

File: Modules/Invoice/Services/InvoiceService.php [hunk 5]
```diff
@@ -81,6 +83,7 @@ public function update($data, $id)
         $invoice->update($data);
         if (isset($data['invoice_file']) and $data['invoice_file']) {
             $this->saveInvoiceFile($invoice, $data['invoice_file']);
+            $this->setInvoiceNumber($invoice);
```
Comment (rathorevaibhav) [inline Modules/Invoice/Services/InvoiceService.php:86]: should this go into InvoiceObserver? As it's happening in both update and delete?

File: Modules/Invoice/Resources/views/index.blade.php [hunk 6]
```diff
@@ -1,93 +1,107 @@
...
+                                                        <input type="hidden" name="year" value="{{ $_GET['year'] }}">
+                                                        <label for="sender_invoice_email">Sender's email adress</label>
+                                                        <input type="email" class="form-control mt-2"
+                                                            name="sender_invoice_email" id="sender_invoice_email"
+                                                            value="{{ config('invoice.pending-invoice-mail.pending-invoice.email') }}">
+                                                        <label for="invoice_email" class="mt-4">CC Email</label>
+                                                        <input type="email" multiple class="form-control mt-2"
+                                                            name="invoice_email" id="invoice_email" placeholder="CC Email"
+                                                            data-toggle="tooltip" data-placement="bottom"
+                                                            title="Please enter a email in comma seprated">
```
Comment (rathorevaibhav) [inline Modules/Invoice/Resources/views/index.blade.php:91]:
```suggestion
                                                            title="Please enter comma separated emails">
```

File: Modules/Invoice/Resources/views/index.blade.php (hunk 3 above)
Comment (pankaj-ag) [reply Modules/Invoice/Resources/views/index.blade.php:10]: I did not make this change. It's old one.

File: Modules/Invoice/Database/Seeders/ResetInvoiceNumberTableSeeder.php (hunk 1 above)
Comment (pankaj-ag) [reply Modules/Invoice/Database/Seeders/ResetInvoiceNumberTableSeeder.php:9]: Make sense. https://github.com/ColoredCow/portal/issues/1378

File: Modules/Invoice/Services/InvoiceService.php (hunk 5 above)
Comment (pankaj-ag) [reply Modules/Invoice/Services/InvoiceService.php:86]: I was also thinking to do that but I was not sure. @P4NK4J will work on this and clean it up.

File: Modules/Invoice/Services/InvoiceService.php (hunk 4 above)
Comment (pankaj-ag) [reply Modules/Invoice/Services/InvoiceService.php]: Done.

File: Modules/Invoice/Services/InvoiceService.php (hunk 5 above)
Comment (rathorevaibhav) [reply Modules/Invoice/Services/InvoiceService.php:88]: No no. I'm talking about this line only. calling the `setInvoiceNumber()` function from observer. I don't think we need to move the whole function there.

Comment (pankaj-ag) [reply Modules/Invoice/Services/InvoiceService.php:88]: Can't do that right now. If you see we are adding the invoice number based on the file path in the database (which is wrong though) and we need to fix that as well.

**Review by rathorevaibhav — APPROVED** [review]

**Pull request merged**